File: packages/one/ios/Nitro/HybridOneUpdates.swift
import NitroModules

// the js face of One.Updates. every call delegates to the launcher, which
// selected this bundle before React Native started: the reads describe the
// running update, check and fetch talk to the update server, and reload
// restarts the host onto the newest ready update.
final class HybridOneUpdates: HybridOneUpdatesSpec {
  var isEnabled: Bool {
    OneUpdatesLauncher.isEnabled
  }

  var runtimeVersion: String? {
    OneUpdatesLauncher.runtimeVersion
  }

  var updateId: String? {
    OneUpdatesLauncher.currentUpdateId()
  }

  var isEmbeddedLaunch: Bool {
    OneUpdatesLauncher.currentManifestJson() == nil
  }

  var createdAt: String? {
    OneUpdatesLauncher.currentCreatedAt()
  }

  var manifestJson: String? {
    OneUpdatesLauncher.currentManifestJson()
  }

  func check() throws -> Promise<OneUpdatesCheckResult> {
    OneUpdatesLauncher.check()
  }

  func fetch() throws -> Promise<OneUpdatesFetchResult> {
    OneUpdatesLauncher.fetch()
  }

  func getStagedJson() throws -> String? {
    OneUpdatesLauncher.currentStagedJson()
  }

  func addStagedListener(listener: @escaping (_ stagedJson: String?) -> Void) throws -> () -> Void {
    OneUpdatesLauncher.addStagedListener(listener)
  }

  func reload() throws -> Promise<Void> {
    OneUpdatesLauncher.reload()
  }
}

File: packages/one/android/src/main/java/com/margelo/nitro/one/HybridOneUpdates.kt
package com.margelo.nitro.one

import com.margelo.nitro.core.Promise

// the js face of One.Updates. every call delegates to the launcher, which
// selected this bundle before React Native started: the reads describe the
// running update, check and fetch talk to the update server, and reload
// restarts the host onto the newest ready update.
class HybridOneUpdates : HybridOneUpdatesSpec() {
    override val isEnabled: Boolean
        get() = OneUpdatesLauncher.isEnabled()

    override val runtimeVersion: String?
        get() = OneUpdatesLauncher.runtimeVersion()

    override val updateId: String?
        get() = OneUpdatesLauncher.currentUpdateId()

    override val isEmbeddedLaunch: Boolean
        get() = OneUpdatesLauncher.currentManifestJson() == null

    override val createdAt: String?
        get() = OneUpdatesLauncher.currentCreatedAt()

    override val manifestJson: String?
        get() = OneUpdatesLauncher.currentManifestJson()

    override fun check(): Promise<OneUpdatesCheckResult> = OneUpdatesLauncher.check()

    override fun fetch(): Promise<OneUpdatesFetchResult> = OneUpdatesLauncher.fetch()

    override fun getStagedJson(): String? = OneUpdatesLauncher.currentStagedJson()

    override fun addStagedListener(listener: (stagedJson: String?) -> Unit): () -> Unit =
        OneUpdatesLauncher.addStagedListener(listener)

    override fun reload(): Promise<Unit> = OneUpdatesLauncher.reload()
}
